### PR TITLE
[docs] Update deployment docs

### PR DESCRIPTION
**Pull request by chunyong-lin**
to: @ryandeivert 
cc: @airbnb/streamalert-maintainers

## Background
Update deployment doc with more information.

## Testing
Locally built docs and the format looks good.

### CODE REVIEW

File: docs/source/deployment.rst [hunk 1]
```diff
@@ -54,6 +60,14 @@ To speed up the Terraform run, the module name may be specified with the ``targe
   $ python manage.py build --target kinesis       # tf_stream_alert_kinesis module
   $ python manage.py build --target stream_alert  # tf_stream_alert module
 
+To apply specific terraform changes, go to ``terraform`` folder and run ``terraform apply`` with ``-target`` option manually:
+
+.. code-block:: bash
+
+  $ cd terraform
```
Comment (ryandeivert) [inline docs/source/deployment.rst]: I'm not sure I'd like to highlight this unless we absolutely have to.

Is there something that `python manage.py list-targets` wouldn't allow you to target with `python manage.py build --target ...`?

**Comment by chunyong-lin** [comment]
@ryandeivert I have addressed  your comment.

File: docs/source/deployment.rst [hunk 2]
```diff
@@ -54,6 +60,29 @@ To speed up the Terraform run, the module name may be specified with the ``targe
   $ python manage.py build --target kinesis       # tf_stream_alert_kinesis module
   $ python manage.py build --target stream_alert  # tf_stream_alert module
 
+To apply specific terraform changes, use ``list-target`` and ``build`` commands following with ``--target`` option:
```
Comment (ryandeivert) [inline docs/source/deployment.rst]: I'm so sorry to make you update this again - but typo here: `list-target` should be `list-targets`

**Comment by ryandeivert** [comment]
thank you so much for these updates!! :)

**Pull request merged**